Cargo install on Windows fails while compiling monotrail-utils

### Steps to Reproduce

1. `cargo install --git https://github.com/mitsuhiko/rye --tag 0.22.0 rye` 

### Expected Result

`rye` is installed via Cargo

### Actual Result

```
   ...
   Compiling monotrail-utils v0.0.1 (https://github.com/konstin/poc-monotrail?rev=136807e1fe87e9319e0223f76b602ba5db881322#136807e1)
error[E0308]: mismatched types
   --> rye\src\cli\add.rs:323:21
    |
322 |                   requirement.version_or_url = Some(VersionOrUrl::VersionSpecifier(
    |                                                     ------------------------------ arguments to this enum variant are incorrect
323 | /                     VersionSpecifiers::from_iter(Some(
324 | |                         VersionSpecifier::new(
325 | |                             // local versions or versions with only one component cannot
326 | |                             // use ~= but need to use ==.
...   |
338 | |                         .map_err(|msg| anyhow!("invalid version specifier: {}", msg))?,
339 | |                     )),
    | |______________________^ expected `VersionSpecifiers`, found a different `VersionSpecifiers`
    |
    = note: `VersionSpecifiers` and `VersionSpecifiers` have similar names, but are actually distinct types
note: `VersionSpecifiers` is defined in crate `pep440_rs`
   --> C:\Users\LeimgruberF\.cargo\registry\src\index.crates.io-6f17d22bba15001f\pep440_rs-0.4.0\src\version_specifier.rs:58:1
    |
58  | pub struct VersionSpecifiers(Vec<VersionSpecifier>);
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
note: `VersionSpecifiers` is defined in crate `pep440_rs`
   --> C:\Users\LeimgruberF\.cargo\registry\src\index.crates.io-6f17d22bba15001f\pep440_rs-0.3.12\src\version_specifier.rs:58:1
    |
58  | pub struct VersionSpecifiers(Vec<VersionSpecifier>);
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    = note: perhaps two different versions of crate `pep440_rs` are being used?
note: tuple variant defined here
   --> C:\Users\LeimgruberF\.cargo\registry\src\index.crates.io-6f17d22bba15001f\pep508_rs-0.2.3\src\lib.rs:371:5
    |
371 |     VersionSpecifier(VersionSpecifiers),
    |     ^^^^^^^^^^^^^^^^

For more information about this error, try `rustc --explain E0308`.
error: could not compile `rye` (bin "rye") due to 1 previous error
error: failed to compile `rye v0.22.0 (https://github.com/mitsuhiko/rye?tag=0.22.0#203a0f33)`, intermediate artifacts can be found at `C:\Users\LEIMGR~1\AppData\Local\Temp\cargo-installN44umq`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
```

### Version Info

OS: Windows
Rust toolchain: stable-x86_64-pc-windows-msvc (default), rustc 1.76.0 (07dca489a 2024-02-04)

### Stacktrace

_No response_

--- a/rye/Cargo.toml
+++ b/rye/Cargo.toml
@@ -33,7 +33,7 @@ nix = { version = "0.27.1", default-features = false, features = ["process"] }
 once_cell = "1.17.1"
 pathdiff = "0.2.1"
 pep440_rs = "0.4.0"
-pep508_rs = "0.2.1"
+pep508_rs = "0.3.0"
 regex = "1.8.1"
 same-file = "1.0.6"
 serde = { version = "1.0.160", features = ["derive"] }
@@ -55,7 +55,7 @@ bzip2 = "0.4.4"
 zip = { version = "0.6.5", features = ["deflate"], default-features = false }
 self-replace = "1.3.5"
 configparser = "3.0.2"
-monotrail-utils = { git = "https://github.com/konstin/poc-monotrail", rev = "136807e1fe87e9319e0223f76b602ba5db881322" }
+monotrail-utils = { git = "https://github.com/konstin/poc-monotrail", rev = "e0251f68c254f834180198b8677fcf85d4b6a844" }
 python-pkginfo = { version = "0.6.0", features = ["serde"] }
 sysinfo = { version = "0.29.4", default-features = false, features = [] }
 home = "0.5.9"

